lib/ProductOpener/Import.pm

```diff
@@ -534,135 +673,7 @@ sub import_csv_file($) {
-						and (defined $org_ref->{gs1_product_name_is_abbreviated}) and ($org_ref->{gs1_product_name_is_abbreviated})) {
-							
-						if ((defined $imported_product_ref->{product_name_fr}) and ($imported_product_ref->{product_name_fr} ne "")) {
-							$imported_product_ref->{abbreviated_product_name_fr} = $imported_product_ref->{product_name_fr};
-							delete $imported_product_ref->{product_name_fr};
-						}
-					}
-					
-					# Nutrition facts marked as "prepared" are in fact for unprepared / as sold product
-					if ((defined $args_ref->{source_id}) and (($args_ref->{source_id} eq "codeonline") or ($args_ref->{source_id} eq "equadis") or ($args_ref->{source_id} eq "agena3000"))
-						and (defined $org_ref->{gs1_nutrients_are_unprepared}) and ($org_ref->{gs1_nutrients_are_unprepared})) {
-						
-						foreach my $field (sort keys %$imported_product_ref) {
-							if ($field =~ /_prepared/) {
-								my $unprepared_field = $` . $';
-								if (((defined $imported_product_ref->{$field}) and ($imported_product_ref->{$field} ne ''))
-									and not ((defined $imported_product_ref->{$unprepared_field}) and ($imported_product_ref->{$unprepared_field} ne "")) ) {
-									$imported_product_ref->{$unprepared_field} = $imported_product_ref->{$field};
-									delete $imported_product_ref->{$field};
-								}
-							}
-						}
-					}	
-				}
-				else {
-					# The org does not exist yet, create it
-
-					defined $stats{orgs_created}{$org_id} or $stats{orgs_created}{$org_id} = 0;
-					$stats{orgs_created}{$org_id}++;
-					
-					$org_ref = create_org($User_id, $org_id);
-					$org_ref->{name} = $imported_product_ref->{org_name};
-
-					# Set the sources field to authorize imports from the source that created the org
-					# e.g. if the org was created by an import of Codeonline or the USDA,
-					# then that source will be able to load new imports automatically
-					# (unless the authorization is revoked by an admin or the org owner)
-
-					if (defined $args_ref->{source_id}) {
-						$org_ref->{"import_source_" . $args_ref->{source_id}} = "on";
-
-						# Check the checkbox for automated exports to the public database
-						$org_ref->{"activate_automated_daily_export_to_public_platform"} = "on";
-					}
-					
-					if (defined $imported_product_ref->{"sources_fields:org-gs1:gln"}) {
-						$org_ref->{sources_field} = {
-							"org-gs1" => {
-								gln => $imported_product_ref->{"sources_fields:org-gs1:gln"}
-							}
-						};
-						if (defined $imported_product_ref->{"sources_fields:org-gs1:partyName"}) {
-							$org_ref->{sources_field}{"org-gs1"}{"partyName"} = $imported_product_ref->{"sources_fields:org-gs1:partyName"};
-						}
-						set_org_gs1_gln($org_ref, $imported_product_ref->{"sources_fields:org-gs1:gln"});
-						$glns_ref = retrieve("$data_root/orgs/orgs_glns.sto");
-					}
-			
-					store_org($org_ref);
-				}
-			}
+			apply_organization_settings_to_product()
```

Comment:
You will need to pass $imported_product_ref and anything the function changes as parameter.

Same thing for $org_id, it's declared in import_csv_file() .

So maybe a first function could be get_org_id_for_product($imported_product_ref);

then apply_organization_settings_to_product($imported_product_ref, $org_id)